File: servlet/3POP_OOP/Human.java

public class Human
{
    int  head=1;  //屬性
    int  eyes=2;
    int  nose=1;
    int  mouth=1;
    int  ears=2;
     //:
     //:

    //方法
    public void  run(int distance)  
    { 
        System.out.println("我跑了" + distance + "公尺");        
    }
    
    public void  jump(int height) 
    { 
          //...  
    }
    
    public void  eat(String food) 
    { 
        System.out.println("我吃了" + food);        
    }
    
    public void  cry( ) 
    { 
          /*... */
    } 

    //建構
    public Human( )  
    {
            /*
             :
             :
            */  
    }

    public Human(String sex)
    {
            /*
             :
             :
            */  
    }
}

class Born	
{
    public static void main(String[] args)  //程式進入點 
	{
		Human Adam = new Human();
		System.out.print("亞當的眼睛有 ");
		System.out.println( Adam.eyes + " 個");
		System.out.println();
            /*
             :
             :
             :
            */  
        Human Eve;
		Eve = new Human("female");
		System.out.print("夏娃說:" );
        Eve.eat("apple");
		System.out.println();
            /*
             :
             :
             :
            */  
    }
}

File: servlet/3POP_OOP/human.cpp

#include <iostream>
#include <cstdlib>
#include <string>

using namespace std;

//物件導向 
class human
{
    private:   
        //屬性
        int head;
        int eyes;
        int nose;
        int ears;
        int mouth;
        
    public:       
        //方法
        void run(int distance)
        {
             cout << "我跑了 " << distance << " 公尺" << endl; 
        }
    
        void jump(int height)
        {
            /*
             *
             */
        }
    
        void eat(string food)
        {
            cout << "The " << food << " is delicious !" << endl;
            //printf("%s%s%s\n","The ",food," is delicious !");
        }
    
        void cry()
        {
            /*
             *
             */
        }
    
        //建構
        human() { /*性別由老天決定*/ }
    
        human(string sex) { /*性別由當事人決定*/ }   
}; 

//程序導向 
void show()
{  
  cout << "Say hi!" << endl; 	
}

//主程式	
int main()  //程式進入點 
{
   human Adam;   
   Adam.run(100);
       
   human *Eve = new human("female");
   Eve->eat("apple");

   system("pause");
   return 0;
}
